pkg/internal/gomod/repo.go

```diff
@@ -0,0 +1,39 @@
+package gomod
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+)
+
+// Repo returns the source code repository URL for the given module import path.
+func Repo(ctx context.Context, importpath string) (string, error) {
```

Comment:
We could add some tests about this function. I added some on the infra-images code.